rasterio/rio/info.py

```diff
@@ -15,16 +15,106 @@
 from rasterio.transform import guard_transform
 
 
+# Handlers for info module options.
+
+def all_handler(ctx, param, value):
+    """Get tags from a template file or command line."""
+    if ctx.obj and ctx.obj.get('like'):
+        value = ctx.obj.get('like')
+    return value
+
+
+def crs_handler(ctx, param, value):
+    """Get crs value from a template file or command line."""
+    if value and ctx.obj and ctx.obj.get('like'):
+        value = ctx.obj.get('like')['crs']
+    elif value:
+        try:
+            value = json.loads(value)
+        except ValueError:
+            pass
+        if not (rasterio.crs.is_geographic_crs(value) or
+                rasterio.crs.is_projected_crs(value)):
+            raise click.BadParameter(
+                "'%s' is not a recognized CRS." % value,
+                param=param, param_hint='crs')
+    return value
+
+
+def like_handler(ctx, param, value):
+    """Copy a dataset's meta property to the command context for access
+    from other callbacks."""
+    if ctx.obj is None:
+        ctx.obj = {}
+    if value:
+        with rasterio.open(value) as src:
+            metadata = src.meta
+            ctx.obj['like'] = metadata
+            ctx.obj['like']['transform'] = metadata['affine']
+            ctx.obj['like']['tags'] = src.tags()
+
+
+def nodata_handler(ctx, param, value):
+    """Get nodata value from a template file or command line."""
+    if value and ctx.obj and ctx.obj.get('like'):
```

Comment:
Both ```if``` and ```elif``` conditions here fail if value is ```0```, which is a valid nodata value.  

Should be ```if value is not None``` instead.